iothub/service/src/IoTHubSasCredential.cs

```diff
@@ -14,33 +14,48 @@ namespace Microsoft.Azure.Devices
     /// Shared access signature credential used to authenticate with IoT hub.
     /// </summary>
     public class IotHubSasCredential
```

Comment:
We are moving away from inheritance and using encapsulation here because we don't want to expose the methods in AzureSasCredential.